Kernel/FileSystem/DevFS.cpp

```diff
@@ -275,8 +275,13 @@ KResultOr<NonnullRefPtr<Inode>> DevFSRootDirectoryInode::create_child(const Stri
         if (name != "pts")
             return EROFS;
         auto new_directory_inode = adopt_ref(*new DevFSPtsDirectoryInode(m_parent_fs));
-        m_subfolders.append(new_directory_inode);
-        m_parent_fs.m_nodes.append(new_directory_inode);
+        if (!m_subfolders.try_append(new_directory_inode))
+            return ENOMEM;
+        if (!m_parent_fs.m_nodes.try_append(new_directory_inode)) {
+            // Undo append above.
+            m_subfolders.take_last();
```

Comment:
Instead of this, can we instead use `try_ensure_capacity` first, which would then guarantee that the append succeeds?

Reply:
Yea, makes sense. I was hoping to eventually ifdef out `AK::Vector<T>::append(..)` and friends in kernel mode... but maybe that's a pipe dream.